contentcuration/kolibri_public/views.py

```diff
@@ -697,8 +698,15 @@ def retrieve(self, request, pk=None):
         :return: an object representing the parent with a pagination object as "children"
         """
 
-        queryset = self.get_tree_queryset(request, pk)
+        try:
+            UUID(pk)
+        except ValueError:
+            return Response(
+                {"error": "Invalid UUID format."},
+                status=status.HTTP_400_BAD_REQUEST
```

Comment:
Again, can 404.

As this is mostly just copy pasted from Kolibri, we could just update the `get_tree_queryset` method to match the Kolibri implementation: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/content/api.py#L1028

(this is useful because it means we continue to keep the two implementations as parallel as possible).